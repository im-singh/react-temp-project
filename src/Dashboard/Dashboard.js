import { Component } from 'react'
import ShowOnMobile from '../Components/ShowOnMobile';

export default class Dashboard extends Component {
    state = {
        isShow: false,
    }
    add = (x, y) => {
        return x + y;
    }
    taxPrice = (x, y) => {
        return 5 + this.add(x, y);
    }
    render() {
        let { x, y, text } = this.props;
        let { isShow } = this.state;
        // console.log("ðŸš€ ~ file: Dashboard.js ~ line 12 ~ Dashboard ~ render ~ x, y ,text", x, y ,text)
        return (
            <div className="App">
                <h1>{text}</h1>
                <p>{this.add(x, y)}</p>
                <p>{this.taxPrice(x, y)}</p>
                {isShow || <p>isShow showing</p>}
                <ShowOnMobile>
                    <h2>Only on mobile</h2>
                </ShowOnMobile>
            </div>
        )
    }
}
