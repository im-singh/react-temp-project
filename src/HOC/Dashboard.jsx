import React from 'react';
import requireAuth from './requireAuth';
const Dashboard = (props) => {
    const handleLogout = () => {
        localStorage.removeItem("isAuth");
        props.history.push("/hoc")
    }
    return (<div>
        Dashboard Page
        <button onClick={handleLogout}>Logout</button>
    </div>);
}

export default requireAuth(Dashboard);