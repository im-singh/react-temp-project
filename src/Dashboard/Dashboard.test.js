import React from 'react';
import Dashboard from './Dashboard';
import renderer from 'react-test-renderer';

test("rendering x+y==x+y correctly", () => {
    let x = 1, y = 2;
    let component = renderer.create(<Dashboard x={x} y={y} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
test("rendering text props correctly", () => {
    let text = 'hello';
    let component = renderer.create(<Dashboard text={text} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})