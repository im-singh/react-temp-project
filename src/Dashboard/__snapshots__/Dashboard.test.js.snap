// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering text props correctly 1`] = `
<div
  className="App"
>
  <h1>
    hello
  </h1>
  <p>
    NaN
  </p>
</div>
`;

exports[`rendering x+y==x+y correctly 1`] = `
<div
  className="App"
>
  <h1 />
  <p>
    3
  </p>
</div>
`;
